From 657230d71f1867a6aa9cc44d628f41b7cf672502 Mon Sep 17 00:00:00 2001
From: Eason Yang <yhyang2@nuvoton.com>
Date: Wed, 8 Feb 2023 13:07:02 +0800
Subject: [PATCH 1/1] Fix firmware version missing at dev tag

The firmware version "v" must at the first character, we should not
handle other "v" (like "dev") to avoid parsing string error.

Signed-off-by: Eason Yang <yhyang2@nuvoton.com>
---
 apphandler.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apphandler.cpp b/apphandler.cpp
index a43200c6..9401c3ea 100644
--- a/apphandler.cpp
+++ b/apphandler.cpp
@@ -490,7 +490,7 @@ int convertVersion(std::string_view s, Revision& rev)
     std::string_view token;
     uint16_t commits;
 
-    auto location = s.find_first_of('v');
+    auto location = s.find_last_of('v', 0);
     if (location != std::string::npos)
     {
         s = s.substr(location + 1);
-- 
2.17.1

